### PR TITLE
fix(platform-config): Rename platform config as extra config

### PR DESCRIPTION
**Pull request by gminn**
fix(platform-config): Rename platform config as extra config

 ### Summary

A new Kconfig option was added to Memfault's Zephyr port,
`CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE=y`, to address
losing a user's platform configurations set in `memfault_platform_config.h`, which
[happened in the nRF Connect SDK recently](https://devzone.nordicsemi.com/f/nordic-q-a/103188/memfault-sdk-integration-missing-user-memfault_platform_config). This commit makes
use of that Kconfig option by renaming the platform config file and including
it in the build if that option is set.

 ### Test Plan

Manually tested `CMakeLists.txt` change:
- Build fails as expected if `CONFIG_MEMFAULT_PLATFORM_EXTRA_CONFIG_FILE=y`,
`CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE=n` (forced this  by
commenting out a `select` line) and NCS platform file is NOT renamed. This
makes sure that the `zephyr_include_directories_ifdef()` for 
`CONFIG_MEMFAULT_METRICS_EXTRA_DEFS_FILE` including the `config/` folder
isn't causing a false positive
- Build succeeds and user `memfault_platform_config.h` settings are included
if same settings as above but NCS file is renamed to `memfault_platform_config_extra.h`.
Successful include of user platform config was checked by setting the heartbeat interval to 1,
and checking for the event overflow within a few seconds of boot

### CODE REVIEW

File: modules/memfault-firmware-sdk/config/memfault_platform_config_extra.h [hunk 1]
```diff
@@ -16,4 +16,4 @@
 
 #ifdef CONFIG_MEMFAULT_NCS_ETB_CAPTURE
 #define MEMFAULT_PLATFORM_FAULT_HANDLER_CUSTOM 1
-#endif
+#endif
```
Comment (ejohnso49) [inline modules/memfault-firmware-sdk/config/memfault_platform_config_extra.h]: Super duper tiny nit: Looks like the end of file newline got removed so now the file looks like it was edited and renamed

**Review by ejohnso49 — APPROVED** [review]
Love it

File: modules/memfault-firmware-sdk/config/memfault_platform_config_extra.h (hunk 1 above)
Comment (gminn) [reply modules/memfault-firmware-sdk/config/memfault_platform_config_extra.h]: Oop my bad, fixed!

**Comment by gminn** [comment]
This has been merged into upstream nrf-sdk!

**Pull request closed without merging**